libs/methodologies/bold/rule-processors/credits/src/rewards-distribution/rewards-distribution.helpers.ts

```diff
@@ -0,0 +1,182 @@
+import { sumBigNumbers } from '@carrot-fndn/shared/helpers';
+import { RewardsDistributionActorType } from '@carrot-fndn/shared/methodologies/bold/types';
+import {
+  type NonEmptyString,
+  type NonZeroPositive,
+} from '@carrot-fndn/shared/types';
+import BigNumber from 'bignumber.js';
+
+import type {
+  ActorsByActorType,
+  AggregateMassIdRewards,
+  Remainder,
+  ResultContentsWithMassIdCertificateValue,
+  RewardsDistribution,
+} from './rewards-distribution.types';
+
+export const formatPercentage = (percentage: BigNumber): BigNumber =>
+  percentage.dividedBy(100);
+
+export const formatDecimalPlaces = (value: BigNumber): BigNumber =>
+  value.decimalPlaces(6, BigNumber.ROUND_DOWN);
+
+export const calculateCreditPercentage = (
+  amount: BigNumber,
+  amountTotal: BigNumber,
+): string =>
+  (amountTotal.isZero()
+    ? new BigNumber(0)
+    : formatDecimalPlaces(amount.dividedBy(amountTotal).multipliedBy(100))
+  ).toString();
+
+export const calculateRemainder = (options: {
+  actors: ActorsByActorType;
+  creditsDocumentUnitPrice: NonZeroPositive;
+  massIdTotalValue: BigNumber;
+}): Remainder => {
+  const { actors, creditsDocumentUnitPrice, massIdTotalValue } = options;
+
+  let participantsAmount = new BigNumber(0);
+  let participantsPercentage = new BigNumber(0);
+
+  for (const actor of actors.values()) {
+    participantsAmount = participantsAmount.plus(actor.amount);
+    participantsPercentage = participantsPercentage.plus(actor.percentage);
+  }
+  const rawAmount = massIdTotalValue
+    .multipliedBy(creditsDocumentUnitPrice)
+    .minus(participantsAmount);
+  const rawPercentage = new BigNumber(100).minus(participantsPercentage);
+
+  return {
+    amount: formatDecimalPlaces(BigNumber.maximum(rawAmount, 0)),
+    percentage: formatDecimalPlaces(BigNumber.maximum(rawPercentage, 0)),
+  };
+};
+
+export const calculateAmount = (options: {
+  creditsDocumentUnitPrice: NonZeroPositive;
+  massIdCertificateValue: BigNumber;
+  massIdPercentage: string;
+  participantAmount: string | undefined;
+}): NonEmptyString => {
+  const amount = formatDecimalPlaces(
+    formatPercentage(new BigNumber(options.massIdPercentage))
+      .multipliedBy(options.massIdCertificateValue)
+      .multipliedBy(new BigNumber(options.creditsDocumentUnitPrice)),
+  );
+
+  return amount.plus(options.participantAmount ?? 0).toString();
+};
```

Comment:
I think it would be easier to read if we actually use two helpers.

```suggestion
 export const calculateAbsoluteValue = (options: {
  value: BigNumber;
  percentage: string;
}): NonEmptyString =>
    formatPercentage(new BigNumber(options.massIdPercentage))
      .multipliedBy(options.massIdCertificateValue);

export const addAmount = (options: {
  pricePerUnit: NonZeroPositive;
  value: BigNumber;
  previousAmount: string | undefined;
}): NonEmptyString => {
  const amount = formatDecimalPlaces(
    value.multipliedBy(new BigNumber(options.creditsDocumentUnitPrice)),
  );
  
  return amount.plus(options.previousAmount ?? 0).toString();
}
```